app/src/main/java/org/kiwix/kiwixmobile/main/KiwixSearchWidget.java

```diff
@@ -49,20 +49,23 @@ public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] a
       /** Search Kiwix intent **/
       Intent mainIntent = new Intent(context, MainActivity.class);
       mainIntent.setAction(TEXT_CLICKED);
+      mainIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
       PendingIntent searchPendingIntent =
           PendingIntent.getActivity(context, (int) (System.currentTimeMillis() % Integer.MAX_VALUE),
               mainIntent, 0);
 
       /** Kiwix icon intent to main app **/
       Intent kiwixIconIntent = new Intent(context, MainActivity.class);
       kiwixIconIntent.setAction(ICON_CLICKED);
+      mainIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
```

Comment:
Shouldn't this be kwixiIconIntent? also same for starIntent below

Reply:
Voice intent is not working!!

Reply:
I think error with the widget is that when we open it again then it shows the problem
please check issue you will get to know #625

Reply:
This code is still the same code duplicated. I would like to see the code look like this
```
views.setOnClickPendingIntent(R.id.search_widget_text, pendingIntent(context, TEXT_CLICKED));

views.setOnClickPendingIntent(R.id.search_widget_icon, pendingIntent(context, ICON_CLICKED));

views.setOnClickPendingIntent(R.id.search_widget_star, pendingIntent(context, MIC_CLICKED));

views.setOnClickPendingIntent(R.id.search_widget_mic, pendingIntent(context, STAR_CLICKED));
```